Classes/Handler/FileUploadHandler.php

```diff
@@ -49,27 +84,28 @@ public function uploadFile(InterestRequestInterface $request): ResponseInterface
             );
         }
 
-        $configuration = $this->objectManager->getConfigurationProvider()->getSettings();
         $httpClient = $this->objectManager->get(Client::class);
         $fileBaseName = basename($data['data']['url']);
-        $storagePath = $configuration['persistence']['fileUploadFolderPath'];
-        [$storageId, $subFolderPath] = explode(':', $storagePath);
-        $storage = $this->objectManager->get(StorageRepository::class)->findByUid((int)$storageId);
 
         if ($data['data']['fileData']) {
             $fileBaseName = $data['data']['name'];
-            $stream = fopen('fileadmin/' . $fileBaseName, 'w');
-            stream_filter_append($stream, 'convert.base64-decode', STREAM_FILTER_WRITE);
-            fwrite($stream, $data['data']['fileData']);
-            fclose($stream);
+            $downloadFolder->createFile($fileBaseName);
+
+            if ($downloadFolder->hasFile($fileBaseName)) {
+                $stream = fopen($downloadFolder->getPublicUrl() . $fileBaseName, 'w');
+                stream_filter_append($stream, 'convert.base64-decode', STREAM_FILTER_WRITE);
+                fwrite($stream, $data['data']['fileData']);
+                fclose($stream);
```

Comment:
You have to use the FAL API to write to this file. You cannot use fwrite directly because the file might not be accessible using a stream.

To still apply the stream filter you can wite to a php://temp stream temporarily and read it into the file using $file->setContents(). https://www.php.net/manual/en/wrappers.php.php#wrappers.php.memory